### ISSUE TITLE
Check overlapping offline segments

### ISSUE THREAD

**Issue by yupeng9** [apache/pinot#5747]
Similar to the missing offline segment checker, it will be useful to have an overlapping segment checker on the intervals to avoid duplicate records.

**Comment by fx19880617** [comment]
Since Pinot expected multiple segments for same day, could you describe more detailed situation of the issue?

Basically how the data segments are organized and how do you find there are duplicates by looking at the segment(name/metadata) (I assume this is not related to looking at real data, since the checker only has access to segment metadata). 

It will help summaries this to a common pattern and design the right way to identify data duplication.

**Comment by yupeng9** [comment]
Right. The generated segment name includes three parts: time column's min value, time columns' max value, sequence id. e.g. `transcript_OFFLINE_1570863600000_1572418800000_0`. 

We have seen some batch ingestion jobs incorrectly generate the records overlapping on the time column. This reflects on the segment metadata, specifically the min/max range of the segments. It will be helpful to have such inbuilt sanity checks to catch this kind of errors.
